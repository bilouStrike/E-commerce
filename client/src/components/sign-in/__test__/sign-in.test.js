import { shallow , mount, spy } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { SignIn } from '../sign-in.component';

describe('Sign In component', () => {

    const mockemailSignInStart = jest.fn();
    const mockgoogleSignInStart = jest.fn();
    const mockHandleSubmit = jest.fn();

    mockHandleSubmit.mockImplementation( async () => {
        mockemailSignInStart('email@email.com', 'password');
    });

    let wrapper = shallow(<SignIn 
        emailSignInStart={mockemailSignInStart} 
        googleSignInStart={mockgoogleSignInStart}/>
    );

    it('expect to render signIn component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('expect exist email input with props', () => {
        expect(wrapper.find('FormInput[type="email"]').props()).toEqual({
            name:"email",
            type:"email",
            handleChange: expect.any(Function) ,
            value:"",
            label:"email",
            required:true
        });
    });
  
    it('expect exist password input with props', () => {
        expect(wrapper.find('FormInput[type="password"]').props()).toEqual({
            name:"password",
            type:"password",
            handleChange: expect.any(Function) ,
            value:"",
            label:"password",
            required:true
        });
    });

    it('expect exist custom button with props', () => {
        expect(wrapper.find('CustomButton[type="button"]').props()).toEqual({
            type:"button",
            onClick: mockgoogleSignInStart,
            isGoogleSignIn:true,
            children: 'Sign in with Google'
        });
    });
    
});
