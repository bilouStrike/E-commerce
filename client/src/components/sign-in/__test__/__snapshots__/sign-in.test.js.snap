// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In component expect to render signIn component 1`] = `
<div
  className="sign-in"
>
  <h2>
    I already have an account
  </h2>
  <span>
    Sign in with your email and password
  </span>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      handleChange={[Function]}
      label="email"
      name="email"
      required={true}
      type="email"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="password"
      name="password"
      required={true}
      type="password"
      value=""
    />
    <div
      className="buttons"
    >
      <CustomButton
        type="submit"
      >
         Sign in 
      </CustomButton>
      <CustomButton
        isGoogleSignIn={true}
        onClick={[MockFunction]}
        type="button"
      >
        Sign in with Google
      </CustomButton>
    </div>
  </form>
</div>
`;
